include/soci/sqlite3/soci-sqlite3.h

```diff
@@ -345,15 +345,23 @@ struct sqlite3_session_backend : details::session_backend
             case db_double:
                 return "real";
             case db_date:
+                return "date";
```

Comment:
This actually changes the affinity of the created column: it used to be `INTEGER` but is now `NUMERIC`. It's not really clear whether this is the right thing to do and if we do decide to do it, it needs to be documented as an incompatible change.

But first of all, could you please explain why do we want/need to do this?

Reply:
date is mapped to db_date but db_date is mapped to integer.
At this point, I am pretty sure that the function `create_column` described in https://soci.sourceforge.net/doc/master/utilities/ was only tested with int.

Reply:
Sorry, I don't understand: this is not about mapping but about the actual SQL sent to SQLite. AFAICS after this change it will use `alter table $table add column $column date` while previously it used `add column integer`. Am I wrong about it?

Reply:
Let's handle this issue separately for now.

Reply:
btw, this is mentioned in #969.